float64 coo_x
float64 coo_y
int32 stop
int32 color